src/iconv.jl

```diff
@@ -240,9 +244,9 @@ end
 ## Functions to encode/decode strings
 
 encoding_string(::Type{ASCIIString}) = "ASCII"
-encoding_string(::Type{UTF8String}) = "UTF-8"
-encoding_string(::Type{UTF16String}) = "UTF-16LE"
-encoding_string(::Type{UTF32String}) = "UTF-32LE"
+encoding_string(::Type{UTF8String})  = "UTF-8"
+encoding_string(::Type{UTF16String}) = (ENDIAN_BOM == 0x04030201) ? "UTF-16LE" : "UTF-16BE"
```

Comment:
Yes, it does. (BTW, I love your typo - "bug-endian" - it totally encapsulates how I feel about having to deal with big-endian platforms!)